docs/proposals/remote-artifacts.md

```diff
@@ -0,0 +1,308 @@
+dealing with container images as well, and this enhancement proposal focuses on adding a remote
+artifacts support to Shipwright's Build API.
+
+# Motivation
+
+Give Shipwright's operator broader build use-case support by enhancing its capabilities to include
+the concept of Artifacts. In other words, remote entities that will be available for the image
+build process.
+
+End users will be able to include remote artifacts as a build runtime dependency, artifacts
+controlled by remote systems will be downloaded before the build process starts. Those artifacts may
+be a pre-compile binary, `jar` files, `war` files, etc.
+
+## Goals
+
+*   Provide means to declare remote artifacts, dependencies that can be employed on builds;
+*   Create the mechanism to download and prepare remote artifacts for builds;
+
+## Non-Goals
+
+*   Automate the upload of local artifacts into the cluster;
+*   Manage remote artifacts;
+*   Amend container images created by the given `BuildStrategy`;
+*   Walking a remote directory tree, similarly to a `git checkout`;
+
+# Proposal
+
+The enhancement proposal is centered around the idea of declaring external dependencies, here
+called "artifacts", and being able to use them during the container image building process.
+
+## Build Artifacts
+
+The remote artifacts will be directly expressed in the body of a `Build` resource, as the
+following example:
+
+```yml
+---
+apiVersion: build.dev/v1alpha1
+kind: Build
+metadata:
+  name: license-file
+spec:
+  sources:
+    - name: license-file
+      type: http
+      url: https://licenses.company.com/customer-id/license.tar
+      http:
+        path: $(workspace)/licenses/license.tar
+```
+
+We will add `sources` section, side by side with current `source`. The idea is to accommodate both
+constructions in `v1alpha1`, and save API breaking changes for upcoming `v1beta1`.
+
+The new `sources` will contain the following attributes:
+
+- `name`: source name (required);
+- `type`: input source type, `git` or `http` (required);
+- `sourceRef`: use a external resource, the name in this field is the resource name (optional);
+- `url`: the URL of the repository or remote artifact (optional);
+- `credentials`: the credentials to interact with the artifact (optional);
+- `http`: settings for `http`, namely `path` (optional);
+- `git`: settings for `git`, namely `revision` (optional);
+
+The resource `UID`/`GID` will be defined by the same user running the Tekton task, later on, we
+can extend the API to support arbitrary configuration.
+
+### Single vs. Multiple Git Repositories
+
+In `v1alpha1` it's expected to inform the Git repository on `.spec.source`, and external artifacts
+informed on `.spec.sources`, therefore, we can address adding this new feature and managing multiple
+Git repositories support in different tracks.
+
+For upcoming work, supporting multiple Git repositories will involve re-defining `/workspace/source`
+location and `$workspace` placeholder.
```

Comment:
Suggestion to clarify this section a bit:

```suggestion
For the initial implementation, we will only support one git repository that is specified in `.spec.source`.
Remote artifacts will be specified n `.spec.sources` only.

We will address supporting multipe Git repositories in a future enhancement proposal. This will involve re-defining `/workspace/source` location and the `$workspace` placeholder.
```